src/Responder/XmlResponder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Responder;
+
+use OAT\SimpleRoster\Generator\MessageIdentifierGenerator;
+use Twig\Environment;
+
+class XmlResponder
+{
+    /** @var Environment */
+    private $twig;
+
+    /** @var MessageIdentifierGenerator */
+    private $messageIdentifierGenerator;
+
+    public function __construct(Environment $twig, MessageIdentifierGenerator $messageIdentifierGenerator)
+    {
+        $this->twig = $twig;
+        $this->messageIdentifierGenerator = $messageIdentifierGenerator;
+    }
+
+    public function createReplaceResultResponse(int $assignmentId): XmlResponse
+    {
+        $xml = $this->twig->render(
+            'basic-outcome/replace-result-response.xml.twig',
```

Comment:
I'd suggest renaming this class to LTIOutcomeResponder to be more specific or to pass the template via a parameter on `createReplaceResultResponse ` to be able to use this class more in a generic way.

Reply:
Thanks for the suggestion. I've renamed the class. Pass the template would be useless because it is too specific. We can re-check it later if we need to return XML responses of other types